src/mscorlib/src/System/String.cs

```diff
@@ -121,30 +113,23 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.EndContractBlock();
 
-            if (separator == null)
-                separator = String.Empty;
-
             using(IEnumerator<T> en = values.GetEnumerator()) {
                 if (!en.MoveNext())
                     return String.Empty;
 
                 StringBuilder result = StringBuilderCache.Acquire();
-                if (en.Current != null) {
-                    // handle the case that the enumeration has null entries
-                    // and the case where their ToString() override is broken
-                    string value = en.Current.ToString();
-                    if (value != null)
-                        result.Append(value);
+                T currentValue = en.Current;
+
+                if (currentValue != null) {
+                    result.Append(currentValue.ToString());
                 }
 
                 while (en.MoveNext()) {
+                    currentValue = en.Current;
```

Comment:
Nit: Should the `currentValue = en.Current;` line be moved after `result.Append(separator);` to be consistent with the previous order of calls (i.e. the separator is appended first, then `en.Current` is accessed)?